src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs

```diff
@@ -413,41 +413,78 @@ public class HttpProtocolTests : HttpClientTestBase
+            if (!PlatformDetection.IsUap)
+            {
+                yield return "HTTP/1.1 200OK";
+                yield return "HTTP/1.1 20c";
+                yield return "HTTP/1.1 23";
+                yield return "HTTP/1.1 2bc";
+            }
+
+            // Skip these test cases on UAP & CurlHandler since the behavior is different.
+            if (!PlatformDetection.IsUap && PlatformDetection.IsWindows)
+            {
+                yield return "NOTHTTP/1.1";
+                yield return "HTTP 1.1 200 OK";
+                yield return "ABCD/1.1 200 OK";
+                yield return "HTTP/1.1";
+                yield return "HTTP\\1.1 200 OK";
+                yield return "NOTHTTP/1.1 200 OK";
+            }
+        }
+        
+        public static TheoryData InvalidStatusLine = GetInvalidStatusLine().ToTheoryData();
+
         [Theory]
-        [InlineData("HTTP/1.1 2345")]
-        [InlineData("HTTP/A.1 200 OK")]
-        [InlineData("HTTP/X.Y.Z 200 OK")]
-        // TODO #24713: The following pass on Windows on .NET Core but fail on .NET Framework.
-        //[InlineData("HTTP/0.1 200 OK")]
-        //[InlineData("HTTP/3.5 200 OK")]
-        //[InlineData("HTTP/1.12 200 OK")]
-        //[InlineData("HTTP/12.1 200 OK")]
-        // TODO #24713: The following pass on Windows on .NET Core but fail on UWP / WinRT.
-        //[InlineData("HTTP/1.1 200 O\nK")]
-        //[InlineData("HTTP/1.1 200OK")]
-        //[InlineData("HTTP/1.1 20c")]
-        //[InlineData("HTTP/1.1 23")]
-        //[InlineData("HTTP/1.1 2bc")]
-        // TODO #24713: The following pass on Windows but fail on CurlHandler on Linux.
-        //[InlineData("NOTHTTP/1.1")]
-        //[InlineData("HTTP/1.A 200 OK")]
-        //[InlineData("HTTP 1.1 200 OK")]
-        //[InlineData("ABCD/1.1 200 OK")]
-        //[InlineData("HTTP/1.1")]
-        //[InlineData("HTTP\\1.1 200 OK")]
-        //[InlineData("HTTP/1.1 ")]
-        //[InlineData("HTTP/1.1 !11")]
-        //[InlineData("HTTP/1.1 a11")]
-        //[InlineData("HTTP/1.1 abc")]
-        //[InlineData("HTTP/1.1 200 O\rK")]
-        //[InlineData("HTTP/1.1\t\t")]
-        //[InlineData("HTTP/1.1\t")]
-        //[InlineData("HTTP/1.1  ")]
-        //[InlineData("NOTHTTP/1.1 200 OK")]
+        [MemberData(nameof(InvalidStatusLine))]
         public async Task GetAsync_InvalidStatusLine_ThrowsException(string responseString)
         {
             await GetAsyncThrowsExceptionHelper(responseString);
         }
+        
+        [Theory]
+        [InlineData("HTTP/1.1 200 O\nK")]
+        public async Task GetAsync_InvalidStatusLine_ThrowsExceptionLFReasonPhrase(string responseString)
+        {
+            if (IsWinHttpHandler || IsNetfxHandler || IsCurlHandler)
+            {
+                await GetAsyncThrowsExceptionHelper(responseString);
+            }
+        }
+        
+        [Theory]
+        [InlineData("HTTP/1.1 200 O\rK")]
```

Comment:
I would move the test case back where it was and just disable it against the bug you filed.